### PR TITLE
Ak admin view items

### LINKED ISSUE #115: Story 20: Admin views all Items
As an Admin
     When I visit "/admin/dashboard"
     Then I should see a link for viewing all items
     And when I click that link
     Then my current path should be "/admin/items"
     Then I should see a table with all items (active and inactive)
     And each item should have:
       - A thumbnail of the image
       - Title that links to the item
       - Description
       - Status
       - Actions (Edit)

### PR DESCRIPTION
**Pull request by akintner**
add admin items route
add admin items_controller
add three new rspec tests, all passing

make admin_items view with table for all items
update admin dashboard with links to admin_items

closes #115

### CODE REVIEW

**Comment by akintner** [comment]
ready for review @wlffann @kheppenstall @tmikeschu

File: app/controllers/admin/items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Admin::ItemsController < Admin::BaseController
+  before_action :require_admin
```
Comment (wlffann) [inline app/controllers/admin/items_controller.rb]: I think if you're inheriting from the admin base controller, you don't need to have before_action here.

Comment (akintner) [inline app/controllers/admin/items_controller.rb]: ah, good catch

**Comment by wlffann** [comment]
Looks, good @akintner ! I left a comment about a redundancy but looks good to merge other than that!

File: app/controllers/admin/items_controller.rb (hunk 1 above)
Comment (akintner) [inline app/controllers/admin/items_controller.rb]: fixed!

File: app/controllers/admin/base_controller.rb [hunk 2]
```diff
@@ -1,5 +1,4 @@
 class Admin::BaseController < ApplicationController
-
```
Comment (tmikeschu) [inline app/controllers/admin/base_controller.rb]: It's the little things that count ;)

File: app/views/admin/items/index.html.haml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+%h1 View All Items
```
Comment (tmikeschu) [inline app/views/admin/items/index.html.haml]: Waiting to style these?

File: spec/features/admin/admin_views_all_items_spec.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      @items = create_list(:item, 4)
+      login_user(admin)
+
+      visit admin_items_path
+
+      expect(page).to have_content 'Item Picture'
+      expect(page).to have_content 'Item Info'
+      expect(page).to have_button 'Edit'
+
+      @items.each do |item|
```
Comment (tmikeschu) [inline spec/features/admin/admin_views_all_items_spec.rb]: Nice use of an each block!

File: spec/features/admin/admin_views_all_items_spec.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      admin = create(:user, role: 1)
+      login_user(admin)
+
+      visit admin_dashboard_path
+      click_link 'View All Items'
+
+      expect(current_path).to eq(admin_items_path)
+    end
+
+    scenario 'can view all items in table' do
```
Comment (tmikeschu) [inline spec/features/admin/admin_views_all_items_spec.rb]: Worth getting the low hanging fruit?

File: app/views/admin/items/index.html.haml (hunk 3 above)
Comment (akintner) [inline app/views/admin/items/index.html.haml]: Yeah, I didn't add a stylesheet for the admin_items

Comment (akintner) [inline app/views/admin/items/index.html.haml]: I'll add another waffle card for that and mess with it tonight.

**Comment by kheppenstall** [comment]
👍 Merging now.

**Pull request merged**